common/djangoapps/third_party_auth/signals/handlers.py

```diff
@@ -0,0 +1,81 @@
+"""
+Signal handlers for third_party_auth app.
+"""
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from edx_django_utils.monitoring import set_custom_attribute
+
+from ..models import SAMLConfiguration, SAMLProviderConfig, ENABLE_SAML_CONFIG_SIGNAL_HANDLERS
+
+
+@receiver(post_save, sender=SAMLConfiguration)
+def update_saml_provider_configs_on_configuration_change(sender, instance, created, **kwargs):
+    """
+    Signal handler to update SAMLProviderConfig instances when SAMLConfiguration is updated.
+
+    When a SAMLConfiguration is updated, ConfigurationModel creates a new version.
+    This handler ensures that all EXISTING SAMLProviderConfig instances that were using
+    the old configuration are updated to point to the new version - NO new providers are created.
+
+    This behavior is controlled by the ENABLE_SAML_CONFIG_SIGNAL_HANDLERS toggle.
+    When disabled, this handler does nothing (legacy behavior).
+
+    Observability is provided regardless of toggle state.
+    """
+    # Check if the toggle is enabled and execute accordingly
```

Comment:
Removing unneeded comment.
```suggestion
    # Check if the toggle is enabled and execute accordingly
```